pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -142,28 +176,42 @@ public void close() {
       dataBuffer.close();
     }
     dataBuffers.clear();
-    headerBuffer.close();
+    for (PinotDataBuffer headerBuffer : headerBuffers) {
```

Comment:
Calling close on closed PinotByteBuffer is safe. We should call close() to ensure all writer resources are correctly released